app/Http/Controllers/Frontend/HotelController.php

```diff
@@ -31,13 +34,130 @@ public function index()
+                                ->orderby('min_price_room', 'ASC');
+                case Hotel::PRICE_DESC:
+                   /**
+                     * Arange hotel by most expensive room in hotels order by decrease
+                     */
+                    return $query->join(\DB::raw("(SELECT hotel_id, max(price) AS max_price_room FROM rooms group by hotel_id) AS most_expensive_rooms"), 'most_expensive_rooms.hotel_id', '=', 'hotels.id')
+                                ->orderby('max_price_room', 'DESC');
+                case Hotel::STAR_ASC:
+                    /**
+                     * Arange hotel by star of hotel order by increase
+                     */
+                    return $query->orderby('star', 'ASC');
+                case Hotel::STAR_DESC:
+                    /**
+                     * Arange hotel by star of hotel order by increase
+                     */
+                    return $query->orderby('star', 'DESC');
+                case Hotel::RATING_DESC:
+                    /**
+                     * Arange hotel by average rating of hotel order by decrease
+                     */
+                    return $query->join(\DB::raw("(SELECT hotel_id, avg(total_rating) AS avg_rating FROM rating_comments group by hotel_id) AS summary_ratings"), 'summary_ratings.hotel_id', '=', 'hotels.id')
+                                ->orderby('avg_rating', 'DESC');
+            }
+        }
+        /**
+         * No arange
+         */
+        return $query;
+    }
+
+    /**
+     * Get query after check checkin rooms of hotels
+     *
+     * @param SearchHotelRequest $request request to display
+     * @param string             $query   query to change
+     * @param array              $columns array of columns
+     *
+     * @return string
+     */
+    private function checkinCondition($request, $query, $columns)
+    {
+        if ($request->has('checkin')) {
+            $checkin = Carbon::createFromFormat(config('showitem.format_datetime'), $request->checkin . " 14:00:00")
+                        ->toDateTimeString();
+            $checkout = Carbon::createFromFormat(config('showitem.format_datetime'), $request->checkin . " 11:00:00")
+                        ->addDay($request->duration)
+                        ->toDateTimeString();
+            /**
+             * Hotel have blank room from checkin day to checkout day
+             */
+            return $query->join('rooms', 'rooms.hotel_id', '=', 'hotels.id')
+                        ->leftJoin(\DB::raw("(SELECT busy_reservations.room_id as room_id, SUM(busy_reservations.quantity) as quantity_busy_reservation FROM (SELECT * FROM reservations WHERE (status = 1) AND ((checkin_date < ? AND checkout_date > ?) OR (checkin_date < ? AND checkout_date > ?))) AS busy_reservations GROUP BY busy_reservations.room_id)  AS busy_rooms"), 'busy_rooms.room_id', '=', 'rooms.id')
```

Comment:
Status này cùng constant truyền vào luôn nha! A nghĩ nó nên là 2 statusm dùng IN